tokenizers/src/tokenizer/mod.rs

```diff
@@ -119,9 +119,13 @@ impl dyn PostProcessor {
     }
 }
 
-/// A `Decoder` has the responsibility to merge the given `Vec<String>` in a `String`.
+/// A `Decoder` changes the raw tokens into its more readable form.
 pub trait Decoder {
-    fn decode(&self, tokens: Vec<String>) -> Result<String>;
+    fn decode(&self, tokens: Vec<String>) -> Result<String> {
+        let results = self.decode_chain(tokens)?;
+        Ok(results.join(""))
```

Comment:
will this line cause any problem for the SequenceDecoder? 
Specifically, one unlikekly edge-case I'm thinking is: one of the sequence decoders is another sequence decoder, and `Ok(results.join(""))` will cause an informaition loss?

as a solution, we can implement `fn decode` for Sequence Decoder?

Reply:
It's OK I think.

The parent `Sequence` would call the child `Sequence.decode_chain` NOT the `decode` function, so we're good, no ?

I could add a test to make sure.

Reply:
ooh I see, so it is no problem in that case